sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -463,4 +479,244 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
     }
     s"$nodeName(${args.mkString(",")})"
   }
+
+  def toJSON: String = {
+    pretty(render(jsonValue))
```

Comment:
We probably don't want this to be pretty by default.  Thats going to make it harder to parse in Spark SQL.